### PR TITLE
LRQA-52673 Added new test for Minium flow

### PR DESCRIPTION
**Pull request by lorenzolapescara**
https://issues.liferay.com/browse/LRQA-52673

### CODE REVIEW

**Comment by iovaneale** [comment]
ci:test

**Review by iovaneale — CHANGES_REQUESTED** [review]
Please change FrontStore.macro into CommerceStoreFront.macro

File: commerce-product-test/src/testFunctional/macros/FrontStore.macro [hunk 1]
```diff
@@ -1,5 +1,20 @@
 definition {
 
+	macro addInitializerProductToCart {
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/macros/FrontStore.macro:3]: Rename: addProductToCart

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+</head>
+<body>
+<table cellpadding="1" cellspacing="1" border="1">
+<thead>
+<tr><td rowspan="1" colspan="3">CommerceMinium</td></tr>
+</thead>
+<tbody>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ACCOUNT_ENTRY</td>
+	<td>//div[@class='row']//..//span[contains(.,'${key_accountName}')]</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:13]: //div[@class='row']//span[contains(.,'${key_accountName}')]

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_ADDRESS_BUTTON</td>
+	<td>//button[contains(.,'Add Address')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_MODAL_ADDRESS</td>
+	<td>//div/label[contains(.,'Address')]//..//input | //div/label[contains(.,'Street')]//..//input</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:28]: //div/label[contains(.,'Address') or contains(.,'Street')]//..//input

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_ADDRESS_ARROW</td>
+	<td>//div[contains(@class,'__actions')]/button</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:33]: //div[contains(@class,'actions')]/button

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_MODAL_NAME</td>
+	<td>//div/label[contains(.,'Name')]//..//input</td>
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_MODAL_PLUS_BUTTON</td>
+	<td>//div/div//span/span[text()='${key_userEmailAddress}']//..//..//..//button</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:53]: //span[text()='${key_userEmailAddress}']//following::button[1]

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_MODAL_REGION</td>
+	<td>//div/label[contains(.,'Region')]//..//select</td>
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_MODAL_TELEPHONE_NUMBER</td>
+	<td>//div/label[contains(.,'Telephone')]//..//input | //div/label[contains(.,'Phone Number')]//..//input</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:63]: //label[contains(.,'Telephone') or contains(.,'Phone Number')]//..//input

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_MODAL_TYPE</td>
+	<td>//div/label[contains(.,'Type')]//..//select</td>
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_MANAGEMENT_ADD_MODAL_ZIP_CODE</td>
+	<td>//div/label[contains(.,'Zip Code')]//..//input | //div/label[contains(.,'Postal Code')]//..//input</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:73]: //label[contains(.,'Zip Code') or contains(.,'Postal Code')]//..//input

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_SELECTOR_SELECT_ACCOUNT</td>
+	<td>//span[contains(text(),'${key_accountName}')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>ACCOUNT_SELECTOR_VIEW_ALL_ACCOUNTS</td>
+	<td>//div[@id='account-manager-curtain']/div/div[2]/div[3]/a</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:138]: //div[@id='account-manager-curtain']//a[contains(.,'View All Accounts')]

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ADD_TO_CART_ARROW_BUTTON</td>
+	<td>//div/p[contains(.,'${key_productSKU}')]/..//div[contains(@class,'add-to-cart-button')]//div[contains(@class,'fwd')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>ADD_TO_CART_BUTTON</td>
+	<td>//div/p[contains(.,'${key_productSKU}')]/..//div//span[contains(.,'${key_productPrice}')]/../..//div[contains(@class,'add-to-cart-button')]</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:153]: //a[contains(.,'${key_productName}')]//..//div[contains(@class,'add-to-cart-button')]

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>CHECKOUT_BUTTON</td>
+	<td>//a[contains(.,'Checkout')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>CHECKOUT_DELIVERY_RADIO</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:162]: CHECKOUT_SHIPPING_OPTION_RADIO

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//label[contains(.,'${key_radioOption}')]/input[@type='radio' and contains(@name,'${key_fieldName}')]</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:163]: //label[contains(.,'${key_radioOption}')]/input[@type='radio']

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>CHECKOUT_SHIPPING_ADDRESS_SELECT</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:167]: Useless. Use Select function.

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//div/label[contains(.,'Choose Shipping Address')]/../select</td>
+	<td></td>
+</tr>
+<tr>
+	<td>COMMERCE_QUANTITY_SELECTOR_INPUT</td>
+	<td>//input[contains(@class,'commerce-quantity-selector__input')]</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:173]: //a[contains(.,'${key_productName}')]//..//div[(@class='commerce-quantity-selector')]//input

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//div[contains(@class,'mini-cart-wrapper')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>MODAL_CREATE_BUTTON</td>
+	<td>//button[contains(.,'Create')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>REVIEW_ORDER_ENTRY</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:187]: PENDING_ORDERS_ENTRY

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//td[contains(., '${key_accountName}')]/../td[contains(.,'${key_orderAuthor}')]/../td/p/span</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:188]: //td[contains(.,'${key_orderAuthor}')]/../td/p/span

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>PENDING_ORDERS_ENTRY_ACCOUNT_NAME</td>
+	<td>//td[contains(., '${key_accountName}')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>PENDING_ORDERS_INACTIVE_PRICE</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:197]: Delete

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//table/tbody//td[3]//span[contains(@class,'--inactive')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>PENDING_ORDERS_PRICE</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:202]: PENDING_ORDER_DETAILS_PRICE

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//table/tbody//td[3]//span[@class='price__value']</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:203]: //tbody//span[@class='price__value']

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>PENDING_ORDERS_PROMO_PRICE</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:207]: DELETE

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//table/tbody//td[3]//span[contains(@class,'promo-price')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>PENDING_ORDERS_QUANTITY</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:212]: PENDING_ORDER_DETAILS_QUANTITY

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 21]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//table/tbody//td[5]//span</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:213]: //tbody//td[5]//span

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 22]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>PENDING_ORDERS_SKU</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:217]: PENDING_ORDER_DETAILS_SKU

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 23]
```diff
@@ -0,0 +1,239 @@
...
+	<td>//table/tbody//td[1]//span[contains(@class,'__value')]</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:218]: //tbody//span[contains(@class,'cell__value')]

File: commerce-product-test/src/testFunctional/paths/CommerceMinium.path [hunk 24]
```diff
@@ -0,0 +1,239 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ROLES_MODAL_ROLE_CHECKBOX</td>
+	<td>//div/input[@type='checkbox' and @name='${key_role}']</td>
+	<td></td>
+</tr>
+<tr>
+	<td>SIDEBAR_MENU_ITEM</td>
+	<td>//div[@class='minium-sidebar']//..//a[@class='main-link' and contains (.,'${key_menuItem}')]</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceMinium.path:228]: //div[@class='minium-sidebar']//..//a[contains (.,'${key_menuItem}')]

File: commerce-product-test/src/testFunctional/paths/CommerceEntry.path [hunk 25]
```diff
@@ -358,6 +358,11 @@
 	<td>//button[contains(@id,'setParentCommercePriceList')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>ADMIN_ACCOUNT_ELLIPSIS</td>
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/paths/CommerceEntry.path:362]: Use Icon#ROW_VERTICAL_ELLIPSIS

File: commerce-product-test/src/testFunctional/tests/CommerceInitializers.testcase [hunk 26]
```diff
@@ -0,0 +1,129 @@
...
+		}
+	}
+
+	test MiniumSiteFullFlow {
+
+		CommerceAccelerators.initializeNewMinium(siteName = "Minium");
+
+		ProductMenu.gotoSiteViaLink(site = "Minium");
+
+		Navigator.openSpecificURL(url = "http://localhost:8080/group/Minium");
```
Comment (iovaneale) [inline commerce-product-test/src/testFunctional/tests/CommerceInitializers.testcase]: Remove it

**Comment by iovaneale** [comment]
ci:test

**Comment by iovaneale** [comment]
ci:test

**Comment by lorenzolapescara** [comment]
ci:test

**Pull request closed without merging**